TTS/tts/datasets/__init__.py

```diff
@@ -30,7 +30,7 @@ def split_dataset(items):
     return items[:eval_split_size], items[eval_split_size:]
 
 
-def load_meta_data(datasets, eval_split=True):
+def load_meta_data(datasets, eval_split=True, ignore_generated_eval=False):
```

Comment:
Honestly, I still don't understand why do you need ìgnore_generated_eval`


If you set eval_split=False you do the same thing as setting `ignore_generated_eval=true`

Reply:
yes but we need to compute the eval_split=True otherwise the eval specified in config.json will be ignored and we need these embeddings during training. Here we need to prevent generating an eval (ignore_generated_eval=True) and at the same time compute the eval  specified in config.json

Reply:
But they don't run at the same time. It either uses the specified eval split or computes an eval split. But they all run if `eval_split`is set `True`. 

So your flag does not change anything.

Reply:
I think you still don't understand the purpose of the flag. The flag works for its purpose which is to prevent the dataset from being split into training and validation (call of the split_dataset function) during the extraction of spectrograms and speaker embeddings.

if eval_split=True and ignore_generated_eval=True and meta_file_val=None (ie a validation csv was not specified in config) this will ignore the split_dataset function call (thus it will not split the dataset into training and validation).

To extract spectrograms and speaker embeddings, it is necessary to ignore the split_dataset function call. This function can remove samples during dataset split between training and validation (in the case of multispeaker training). If samples are removed when extracting embeddings the training will break (since the embedding of that sample will be missing).

We cannot simply pass eval_split=False because for the speaker embeddings extraction the validation set is necessary (since validation will be used in training, if we do not extract the speaker embeddings from these samples the training will break !). 

So if we extract the speaker embeddings with eval_split=False and the user specifies validation in the configuration file ( meta_file_val != None). When the user starts training with these speaker embeddings, the training will break during the validation, because the speakers.json does not have the speaker embeddings from the validation set.

Reply:
if you set `eval_split=false` there is no eval split and you don't need to prevent it since it does not exist. 

So eval_split=True and ignore_generated_eval=True and meta_file_val=None is equal to eval_split=False

Reply:
If you set eval_split=False, there is no eval split and all the dataset is regarded as training data and all the d-vectors will be extracted anyways.

Reply:
Why don't you set meta_file_val to None in the compute_speaker_embedding rather than adding an argument to save the situation in somewhere else ?

Reply:
Indeed. However we have no control (in the extraction script) over whether meta_file_val is specified (meta_file_val!=None) or not (meta_file_val=None) . So we need a flag, in case the user doesn't specify an eval ( meta_file_val==None) we ignore the problem of possible loss of samples caused by splitting the dataset (split_dataset function call) and this is where the flag was made !

Reply:
> Why don't you set meta_file_val to None in the compute_speaker_embedding rather than adding an argument to save the situation in somewhere else ?

It has datasets that have validation and training in separate CSVs. This way we would force the user to change config.json just to extract the speaker embeddings. Forcing him to put the validation file as training in config.json. And this can cause confusion! The purpose of the compute_embeddings script is for the user to use the config.json used for training and just run the script getting the result and a speakers.json ready for training. The flag is the easiest way to enable this without extra work on the user's part!

Reply:
So this does not solve the problem in `compute_embeddings.py`?
 
```python
train_items, eval_items = load_meta_data(dataset_config, eval_split=False)
data_items = train_items + eval_items
```

Reply:
In my branch it's ok: https://github.com/Edresson/Coqui-TTS/blob/dev/TTS/bin/compute_embeddings.py#L35

make sure you are not looking at the wrong commit.

Reply:
I am confused :) ?
What do you mean it's ok? 
Does the change I suggested above solve the issue or not?

Reply:
oh ok sorry my bad.

 No it's not solve. As I explained above we need to ignore the generated validation. Because if we use the split_dataset function to split the training and validation this can cause loss of samples when the dataset is multispeaker.

Reply:
you don't lose any samples above. We merge eval and train samples.

Reply:
you still don't understand the problem! It is necessary to skip the eval split generation when the user does not provide a CSV for validation !. The split_dataset function that splits the dataset in training and validation can in a multispeaker scenario remove samples (please see here: https://github.com/Edresson/Coqui-TTS/blob/d906fea08cf96c95d78fa01cdc935fc6a87c6685/TTS/tts/datasets___init .py#L28). So we need to ignore the split_dataset function call and not generate an eval_split if the user doesn't specify it. I've already added eval+training, but that doesn't solve it because the split_dataset function can remove samples in multispeaker scenarios.

Reply:
ok, I found the problem. Apparently it was some old bug in the function now everything works fine (after merges). If this happens again I can see how to solve it and send another PR. So at the moment I tried to reproduce the problem after the merge and it doesn't happen anymore.,